File: Frontend/reddit/src/app/model/post.ts
export interface Post {
    id: number;
    title: string;
    text: string;
    creationDate: Date;
    imagePath: string | null;
    userId: number;
    flairId: number | null;
    communityId: number;
    // reaction: [key:number, value: Reaction];
    // flair: Falir;
    // community: Community;
    // user: User;
    // comment: [key:number, value: Comment];
}

File: Android/Reddit/app/src/main/java/com/ognjenlazic/reddit/model/Post.kt
package com.ognjenlazic.reddit.model

import com.ognjenlazic.reddit.network.model.UserResponse

data class Post(
    val communityId: Int,
    val creationDate: List<Int>,
    val flairId: Int,
    val id: Int,
    val imagePath: String?,
    var text: String,
    var title: String,
    val userId: Int,
    var user: UserResponse
){
    constructor(): this(
        0, emptyList(), 0, 0, "", "", "", 0, UserResponse()
    )
}